### ISSUE TITLE
Images missing alt text

### ISSUE THREAD

**Issue by AymenLoukil** [HTTPArchive/almanac.httparchive.org#379]
https://almanac.httparchive.org/static/images/character-markup.png
https://almanac.httparchive.org/static/images/methodology-characters.png
https://almanac.httparchive.org/static/images/character-star.png
https://almanac.httparchive.org/static/images/character-hat.png
https://almanac.httparchive.org/static/images/menu.png
https://almanac.httparchive.org/static/images/twitter.png
https://almanac.httparchive.org/static/images/methodology-banner.png
https://almanac.httparchive.org/static/images/character-file.png
https://almanac.httparchive.org/static/images/character-painter.png
https://almanac.httparchive.org/static/images/avatars/1.jpg
https://almanac.httparchive.org/static/images/avatars/8.jpg
https://almanac.httparchive.org/static/images/avatars/4.jpg
https://almanac.httparchive.org/static/images/avatars/9.jpg
https://almanac.httparchive.org/static/images/avatars/0.jpg
https://almanac.httparchive.org/static/images/avatars/11.jpg
https://almanac.httparchive.org/static/images/2019/05_Third_Parties/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/03_Markup/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/02_CSS/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/14_CMS/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/13_Ecommerce/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/19_Resource_Hints/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/09_Accessibility/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/07_Performance/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/07_Performance/fig7.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig11.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig2.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig10.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig12.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig16.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig1.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig6.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig5.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig15.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig18.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig14.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig17.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig3.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig4.png
https://almanac.httparchive.org/static/images/2019/07_Performance/fig13.png
https://almanac.httparchive.org/static/images/2019/20_HTTP_2/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/15_Compression/ch15_fig3_compressed_content_types_filtered.jpg
https://almanac.httparchive.org/static/images/2019/15_Compression/ch15_fig5_compression_by_content_type_mobile.jpg
https://almanac.httparchive.org/static/images/2019/15_Compression/ch15_fig10_lighthouse_compression_byte_savings.jpg
https://almanac.httparchive.org/static/images/2019/15_Compression/ch15_fig9_lighthouse_compression_scores.jpg
https://almanac.httparchive.org/static/images/2019/15_Compression/ch15_fig8_lighthouse.jpg
https://almanac.httparchive.org/static/images/2019/15_Compression/ch15_fig6_compression_by_content_type_pct_desktop.jpg
https://almanac.httparchive.org/static/images/2019/15_Compression/fig2.png
https://almanac.httparchive.org/static/images/2019/15_Compression/ch15_fig7_compression_by_content_type_pct_mobile.jpg
https://almanac.httparchive.org/static/images/2019/15_Compression/ch15_fig4_compression_by_content_type_desktop.jpg
https://almanac.httparchive.org/static/images/2019/16_Caching/fig21.png
https://almanac.httparchive.org/static/images/2019/16_Caching/fig3.png
https://almanac.httparchive.org/static/images/2019/16_Caching/fig12.png
https://almanac.httparchive.org/static/images/2019/16_Caching/fig2.png
https://almanac.httparchive.org/static/images/2019/16_Caching/fig14.png
https://almanac.httparchive.org/static/images/2019/16_Caching/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/16_Caching/fig7.png
https://almanac.httparchive.org/static/images/2019/12_Mobile_Web/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/16_Caching/fig13.png
https://almanac.httparchive.org/static/images/2019/16_Caching/fig6.png
https://almanac.httparchive.org/static/images/2019/16_Caching/fig22.png
https://almanac.httparchive.org/static/images/2019/16_Caching/fig9.png
https://almanac.httparchive.org/static/images/2019/16_Caching/fig5.png
https://almanac.httparchive.org/static/images/2019/01_JavaScript/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/04_Media/hero_lg.jpg
https://almanac.httparchive.org/static/images/2019/11_PWA/hero_lg.jpg

**Comment by rviscomi** [comment]
It'd be helpful to know on which pages these images are used. Also a few of these are for unedited chapters

**Comment by bazzadp** [comment]
Chapter 16 were a lot of these and their fixed now and chapter is live. Could you rerun when you get a second @AymenLoukil and update your inital comment. Looks like a lot are hero images. Not sure what to do with them. Do they count as decorative so don't need it as, unlike the figures, they don't actually have meaning?

**Comment by rviscomi** [comment]
+1 also curious whether heroes need alt text.

**Comment by bazzadp** [comment]
Well I think they are great eye-candy 😀: https://www.w3.org/WAI/tutorials/images/decorative/#image-used-for-ambiance-eye-candy

**Comment by AymenLoukil** [comment]
> Chapter 16 were a lot of these and their fixed now and chapter is live. Could you rerun when you get a second @AymenLoukil and update your inital comment. Looks like a lot are hero images. Not sure what to do with them. Do they count as decorative so don't need it as, unlike the figures, they don't actually have meaning?

My check was on the live version. You pushed to it the modfications ?

**Comment by bazzadp** [comment]
> My check was on the live version. You pushed to it the modfications ?

@rviscomi published it to live this morning.

**Comment by AymenLoukil** [comment]
Ok @bazzadp thank you. I run it now

**Comment by AymenLoukil** [comment]
Updated

**Comment by bazzadp** [comment]
OK it looks worse 😞

However it's not as bad as it looks. We are now using this markup like the example below:

```
      <figure id="fig-21">
        <iframe aria-labelledby="fig21-caption" width="600" height="371" seamless="" frameborder="0" scrolling="no" src="https://docs.google.com/spreadsheets/d/e/2PACX-1vT3GWCs19Wq0mu0zgIlKRc8zcXgmVEk2xFHuzZACiWVtqOv8FO5gfHwBxa0mhU6O9TBY8ODdN4Zjd_O/pubchart?oid=827424070&amp;format=interactive"></iframe>
        <a href="/static/images/2019/16_Caching/fig21.png" class="fig-mobile">
          <img src="/static/images/2019/16_Caching/fig21.png" width="600" aria-labelledby="fig21-caption" />
        </a>
        <div id="fig21-caption" class="visually-hidden">A stacked bar chart 38.2% of websites get a score of &lt; 10%, 29.0% of websites get a score between 10% and 39%, 18.7% of websites get a score of 40%-79%, 10.7% of websites get a score of 80% - 99%, and 3.4% of websites get a score of 100%.</div>
        <figcaption>Figure 21. Distribution of Lighthouse scores for the "Uses Long Cache TTL" audit for mobile web pages.</figcaption>
      </figure>
```

So in this case we do have the descriptive text - just not in an `alt` attribute. Instead it's in a special div: `<div id="fig21-caption" class="visually-hidden">`. Can't really expect the tool to know that, but it'll get worse as we convert more chapters to that format.

I say we close this issue.

**Comment by rviscomi** [comment]
One concern is that `aria-labelledby` works fine for a11y but I'm not sure if search indexers follow it or simply rely on the `alt` attribute.

**Comment by AymenLoukil** [comment]
SE use alt value ofc

**Comment by bazzadp** [comment]
> One concern is that `aria-labelledby` works fine for a11y but I'm not sure if search indexers follow it or simply rely on the `alt` attribute.

I'm not  sure this is the right way to go anyway. I think we should have `alt` tags and use `aria-describedby` rather than `aria-labelledby`. In text we say "as seen in figure 12", so I think the image should have an Alt tag of "Figure 12" (with a bit more text) so it can easily be found, and then the `aria-describedby` can give the full detail.

And that would then also solve this issue.

Thoughts? See an upcoming PR to see what I mean.

**Comment by AymenLoukil** [comment]
The alternative text should describe what the photo is about

**Comment by bazzadp** [comment]
Looks at this example that I'm proposing:

```html
<figure id="fig-13">
  <iframe aria-describedby="fig13-caption" title="Figure 13. What asset types is push used for?" width="600" height="371" seamless="" frameborder="0" scrolling="no" loading="lazy" src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQLxLA5Nojw28P7ceisqti3oTmNSM-HIRIR0bDb2icJS5TzONvRhdqxQcooh_45TmK97XVpot4kEQA0/pubchart?oid=466353517&amp;format=interactive"></iframe>
  <a href="/static/images/2019/20_HTTP_2/ch20_fig13_what_push_is_used_for.png" class="fig-mobile">
    <img src="/static/images/2019/20_HTTP_2/ch20_fig13_what_push_is_used_for.png" alt="Figure 13. What asset types is push used for?" aria-describedby="fig13-caption" width="600" data-width="600" data-height="371" data-seamless="" data-frameborder="0" data-scrolling="no" data-src="https://docs.google.com/spreadsheets/d/e/2PACX-1vQLxLA5Nojw28P7ceisqti3oTmNSM-HIRIR0bDb2icJS5TzONvRhdqxQcooh_45TmK97XVpot4kEQA0/pubchart?oid=466353517&amp;format=interactive" loading="lazy">
  </a>
  <div id="fig13-caption" class="visually-hidden">Pie chart breaking down the percent of asset types that are pushed. JavaScript makes up almost half of the assets, then CSS with about a quarter, images about an eighth, and various text-based types making up the rest.</div>
  <figcaption>Figure 13. What asset types is push used for?</figcaption>
</figure>
```

**Comment by AymenLoukil** [comment]
This one is great :100:

**Comment by bazzadp** [comment]
@AymenLoukil can you rerun this? Just pushed a load of descriptions for the chapters.

**Comment by AymenLoukil** [comment]
Ok @bazzadp

**Comment by AymenLoukil** [comment]
It is much better now. We could close the issue.
